consensus/consensus_v2.go

```diff
@@ -274,11 +303,15 @@ func (consensus *Consensus) Start(
 				}
 			case <-consensus.syncReadyChan:
 				consensus.getLogger().Info().Msg("[ConsensusMainLoop] syncReadyChan")
-				consensus.SetBlockNum(consensus.Blockchain.CurrentHeader().Number().Uint64() + 1)
-				consensus.SetViewIDs(consensus.Blockchain.CurrentHeader().ViewID().Uint64() + 1)
-				mode := consensus.UpdateConsensusInformation()
-				consensus.current.SetMode(mode)
-				consensus.getLogger().Info().Str("Mode", mode.String()).Msg("Node is IN SYNC")
+				consensus.mutex.Lock()
+				if consensus.blockNum < consensus.Blockchain.CurrentHeader().Number().Uint64()+1 {
+					consensus.SetBlockNum(consensus.Blockchain.CurrentHeader().Number().Uint64() + 1)
+					consensus.SetViewIDs(consensus.Blockchain.CurrentHeader().ViewID().Uint64() + 1)
+					mode := consensus.UpdateConsensusInformation()
```

Comment:
that's a good point. Though it's a rare case that the new leader just sync to the latest block. But I also notices the UpdateConsensusInformation at end of sync. Is it purposeful? It thought about removing it. But in case it's for other purpose, I didn't touch it. What do you think?

Reply:
I was thinking about removing it at the end of sync too. But I didn't touch the code just as the same reason as you LOL. My concern is following:

1.  `UpdateConsensusInformation` is called immediately right after sync finished.
2. If we keep it in consensus loop, it could happen that the loop was busy dealing with other tasks and there is a delay of executing `UpdateConsensusInformation`. 

So if we can assure that it's OK to execute `UpdateConsensusInformation` and `SetMode` a bit later after sync finished, we can be confident to say that we can remove it from sync logic.

Reply:
Let's keep it that way to avoid new logic bugs. We can come back to this later.

Reply:
Sure, NP